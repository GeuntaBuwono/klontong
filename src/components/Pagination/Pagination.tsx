import classNames from 'classnames';

type PaginationProps = {
	activePage: number;
	isHasNextPage: boolean;
	onClickPrev: () => void;
	onClickNext: () => void;
};

export const Pagination = ({
	activePage,
	isHasNextPage,
	onClickPrev,
	onClickNext,
}: PaginationProps) => {
	const pageItemClassName = classNames('page-item active');

	return (
		<div className="flex justify-center">
			<nav aria-label="Page navigation example">
				<ul className="flex list-style-none">
					{activePage > 1 && (
						<>
							<li className="page-item">
								<button
									onClick={onClickPrev}
									className="page-link relative block py-1.5 px-3 border-0 bg-transparent outline-none transition-all duration-300 text-gray-800 hover:text-gray-800 hover:bg-gray-200 focus:shadow-none"
								>
									Prev
								</button>
							</li>
							<li className={pageItemClassName}>
								<button
									onClick={onClickPrev}
									className="page-link relative block py-1.5 px-3 border-0 bg-transparent outline-none transition-all duration-300 text-gray-800 hover:text-gray-800 hover:bg-gray-200 focus:shadow-none"
								>
									{activePage - 1}
								</button>
							</li>
						</>
					)}
					<li className={pageItemClassName}>
						<button className="page-link relative block py-1.5 px-3 border-0 bg-blue-600 outline-none transition-all duration-300 text-white hover:text-white hover:bg-blue-600 shadow-md focus:shadow-md">
							{activePage} <span className="visually-hidden"></span>
						</button>
					</li>
					{isHasNextPage && (
						<>
							<li className={pageItemClassName}>
								<button
									onClick={onClickNext}
									className="page-link relative block py-1.5 px-3 border-0 bg-transparent outline-none transition-all duration-300 text-gray-800 hover:text-gray-800 hover:bg-gray-200 focus:shadow-none"
								>
									{activePage + 1}
								</button>
							</li>
							<li className="page-item">
								<button
									onClick={onClickNext}
									className="page-link relative block py-1.5 px-3 border-0 bg-transparent outline-none transition-all duration-300 text-gray-800 hover:text-gray-800 hover:bg-gray-200 focus:shadow-none"
								>
									Next
								</button>
							</li>
						</>
					)}
				</ul>
			</nav>
		</div>
	);
};
