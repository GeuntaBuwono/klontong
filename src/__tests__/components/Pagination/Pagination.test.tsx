import {Pagination} from '@components/Pagination/Pagination';
import {render} from '@testing-library/react';

describe('Pagination', () => {
	const testElement = (
		<Pagination
			onClickPrev={jest.fn()}
			onClickNext={jest.fn()}
			activePage={1}
			isHasNextPage={false}
		/>
	);
	test('snapshot', () => {
		expect(testElement).toMatchSnapshot();
	});

	test('check activePage className', () => {
		render(testElement);
		const testActivePage = document.getElementsByClassName('active');
		expect(testActivePage).toBeTruthy();
	});
});
