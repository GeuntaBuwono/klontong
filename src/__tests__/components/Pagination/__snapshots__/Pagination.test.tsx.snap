// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination snapshot 1`] = `
<Pagination
  activePage={1}
  isHasNextPage={false}
  onClickNext={[MockFunction]}
  onClickPrev={[MockFunction]}
/>
`;
